src/app/admin/createUser/CreateUserForm.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import { CenterComponent, FormErrorText, StyledForm } from "@/components/Form/formStyling";
+import Button from "@mui/material/Button";
+import AccountDetails from "@/app/admin/createUser/createUserFormSections/AccountDetails";
+import UserRole from "@/app/admin/createUser/createUserFormSections/UserRole";
+import {
+    checkErrorOnSubmit,
+    Errors,
+    FormErrors,
+    setError,
+    setField,
+} from "@/components/Form/formFunctions";
+import { createUser } from "@/app/admin/adminActions";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faUserPlus } from "@fortawesome/free-solid-svg-icons";
+import RefreshPageButton from "@/app/admin/RefreshPageButton";
+
+interface CreateUserDetails {
+    email: string;
+    password: string;
+    role: "admin" | "caller";
+}
+
+const initialFields: CreateUserDetails = {
+    email: "",
+    password: "",
+    role: "caller",
+};
+
+const initialFormErrors: FormErrors = {
+    email: Errors.initial,
+    password: Errors.initial,
+    role: Errors.none,
+};
+
+const formSections = [AccountDetails, UserRole];
+
+const CreateUserForm: React.FC<{}> = () => {
+    const [fields, setFields] = useState(initialFields);
+    const [formErrors, setFormErrors] = useState(initialFormErrors);
+
+    const fieldSetter = setField(setFields, fields);
+    const errorSetter = setError(setFormErrors, formErrors);
+
+    const [submitError, setSubmitError] = useState(Errors.none);
+    const [submitErrorMessage, setSubmitErrorMessage] = useState("");
+    const [submitDisabled, setSubmitDisabled] = useState(false);
+
+    const [refreshRequired, setRefreshRequired] = useState(false);
+
+    const submitForm = async (): Promise<void> => {
+        setSubmitDisabled(true);
+
+        if (checkErrorOnSubmit(formErrors, setFormErrors)) {
+            setSubmitError(Errors.submit);
+            setSubmitDisabled(false);
+            return;
+        }
+
+        const response = await createUser(fields);
+
+        if (response.error) {
+            setSubmitError(Errors.external);
+            setSubmitErrorMessage(response.error.message);
```

Comment:
do you unset the submiterror on success?

Reply:
Nope! I noticed a bug with this previously...and proceeded not to check it out and have it fixed. But it's fixed now - good catch